src/mlpack/bindings/R/print_R.cpp

```diff
@@ -0,0 +1,247 @@
+{
+  // Restore parameters.
+  IO::RestoreSettings(doc.programName);
+
+  map<string, util::ParamData>& parameters = IO::Parameters();
+  typedef map<string, util::ParamData>::iterator ParamIter;
+
+  // First, let's get a list of input and output options.  We'll take two passes
+  // so that the required input options are the first in the list.
+  vector<string> inputOptions, outputOptions;
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    util::ParamData& d = it->second;
+    if (d.input && d.required)
+    {
+      // Ignore some parameters.
+      if (d.name != "help" && d.name != "info" &&
+          d.name != "version")
+        inputOptions.push_back(it->first);
+    }
+    else if (!d.input)
+    {
+      outputOptions.push_back(it->first);
+    }
+  }
+
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    util::ParamData& d = it->second;
+    if (d.input && !d.required &&
+        d.name != "help" && d.name != "info" &&
+        d.name != "version")
+      inputOptions.push_back(it->first);
+  }
+
+  // Print the documentation.
+  // Print programName as @title.
+  cout << "#' @title ";
+  cout << util::HyphenateString(doc.programName, "#'   ") << endl;
+  cout << "#'" << endl;
+
+  // Next print the short description as @description.
+  cout << "#' @description" << endl;
+  cout << "#' ";
+  cout << util::HyphenateString(doc.shortDescription, "#' ") << endl;
+
+  // Next, print information on the input options.
+  cout << "#'" << endl;
+
+  for (size_t i = 0; i < inputOptions.size(); ++i)
+  {
+    const string& opt = inputOptions[i];
+    util::ParamData& d = parameters.at(opt);
+
+    bool out = false;
+    IO::GetSingleton().functionMap[d.tname]["PrintDoc"](d, NULL, (void*) &out);
+
+    cout << endl;
+  }
+  cout << "#'" << endl;
+
+  // Next, print information on the output options.
+  cout << "#' @return A list with several components:" << endl;
+
+  for (size_t i = 0; i < outputOptions.size(); ++i)
+  {
+    const string& opt = outputOptions[i];
+    util::ParamData& d = parameters.at(opt);
+
+    bool out = true;
+    IO::GetSingleton().functionMap[d.tname]["PrintDoc"](d, NULL, (void*) &out);
+
+    cout << endl;
+  }
+  cout << "#'" << endl;
+
+  // Next print the long description as @details.
+  cout << "#' @details" << endl;
+  cout << "#' ";
+  cout << util::HyphenateString(doc.longDescription(), "#' ") << endl;
+  cout << "#'" << endl;
+  cout << "#' @author" << endl;
+  cout << "#' mlpack developers" << endl;
+  cout << "#'" << endl;
+
+  // Next print the example as @examples.
+  cout << "#' @export" << endl;
+  if (doc.example.size() != 0)
+    cout << "#' @examples" << endl;
+  for (size_t j = 0; j < doc.example.size(); ++j)
+  {
+    const std::string str = doc.example[j]();
+    size_t pos = 0;
+    while (pos < str.length())
+    {
+      size_t splitpos = 0;
+      // Find where example starts.
+      splitpos = str.find("\n\\donttest{", pos) - 1;
```

Comment:
One more question(I am really sorry too many questions) -- I guess CRAN will not run these examples as none of dataset given in example are valid/available. If it runs the examples should I replace this with `\dontrun{}`.

Example: https://gist.github.com/4967c5939ba054bc868415d1e3e5c956#file-lmnn-r-L133

Reply:
@Yashwants19 we like questions. Keep asking them!

For the question contents itself, yes. We will need to switch to `\dontrun` from `\donttest`. I'm not sure CRAN will like that though. We could maybe make a case that the unit tests from `mlpack` central should cover each individual function. 

Otherwise, I think we'd have to make sure each example was ran on a data set.

Reply:
In general, we don't have support to make each of these examples into something that would directly run, since they don't contain any setup or dataset loading, etc.  If the CRAN maintainers are worried about the tests, the argument that we would need to make is that (a) we have tests ensuring that options of each type make it to mlpack correctly and back (in `src/mlpack/bindings/R/tests/`), and (b) each individual binding is tested thoroughly in a language-agnostic way in `src/mlpack/tests/main_tests/`.  So, if all the types make it to mlpack correctly, then the language-agnostic tests in `src/mlpack/tests/main_tests/` pretty much cover everything.

Some testable examples that we do have are the quickstart examples.

Reply:
Done. I have switch `\donttest` to `\dontrun` in the last commit.